llms/mlx_lm/tuner/dora.py

```diff
@@ -93,3 +100,107 @@ def __call__(self, x):
         if "bias" in self.linear:
             out = out + self.linear.bias
         return out
+
+
+class DoRAEmbedding(nn.Module):
+    def from_embedding(
+        embedding: nn.Embedding,
+        r: int = 8,
+        dropout: float = 0.0,
+        scale: float = 20.0,
+    ):
+        num_embeddings, dims = embedding.weight.shape
+
+        # TODO support quantized weights in DoRALinear
+        if isinstance(embedding, nn.QuantizedLinear):
+            raise ValueError("DoRAEmbedding does not yet support quantization.")
+        dora_embedding = DoRAEmbedding(
+            num_embeddings=num_embeddings,
+            dims=dims,
+            r=r,
+            dropout=dropout,
+            scale=scale,
+        )
+        dora_embedding.set_embedding(embedding)
+        return dora_embedding
+
+    from_base = from_embedding
+
+    def to_embedding(self, de_quantize: bool = False):
+        embedding = self.embedding
+        weight = embedding.weight
+
+        # Use the same type as the linear weight if not quantized
+        dtype = weight.dtype
+
+        num_embeddings, dims = weight.shape
+        fused_embedding = nn.Embedding(num_embeddings, dims)
+
+        lora_a = (self.scale * self.lora_a).astype(dtype)
+        lora_b = self.lora_b.astype(dtype)
+        weight = weight + lora_a @ lora_b
+        norm_scale = self.m / mx.linalg.norm(weight, axis=1)
+        fused_embedding.weight = norm_scale[:, None] * weight
+
+        return fused_embedding
+
+    fuse = to_embedding
```

Comment:
Why don't we just call it `fuse` and get rid of the `to_embedding` and `to_linear` etc. That seems cleaner, no?

Reply:
Agreed!

Reply:
I also changed `from_embedding` and `from_linear` to `from_base`